File: src/main/java/fm/comunas/fm/building/adapter/api/rest/dto/BuildingPartWeightDto.java
package fm.comunas.fm.building.adapter.api.rest.dto;

import fm.comunas.ddd.enums.adapter.api.jsonapi.dto.EnumeratedDto;
import lombok.Builder;
import lombok.Data;

@Data()
@Builder
public class BuildingPartWeightDto {

	private EnumeratedDto part;
	private Integer weight;

	private Integer lifeTime20;
	private Integer lifeTime50;
	private Integer lifeTime70;
	private Integer lifeTime85;
	private Integer lifeTime95;
	private Integer lifeTime100;

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/adapter/rest/dto/BuildingPartWeightDto.kt
package io.zeitwert.fm.building.adapter.rest.dto

import io.zeitwert.app.api.jsonapi.EnumeratedDto

data class BuildingPartWeightDto(
	var part: EnumeratedDto,
	var weight: Int,
	var lifeTime20: Int,
	var lifeTime50: Int,
	var lifeTime70: Int,
	var lifeTime85: Int,
	var lifeTime95: Int,
	var lifeTime100: Int,
)
